com.hazelcast.jet.impl.connector.StreamJmsPTest.when_queue

https://hazelcast-l337.ci.cloudbees.com/view/Jet/job/jet/com.hazelcast.jet$hazelcast-jet-core/843/testReport/junit/com.hazelcast.jet.impl.connector/StreamJmsPTest/when_queue/

```
Error Message
expected:<34960dc2-c254-4442-89f5-b6aaa98d70c3> but was:<null>
Stacktrace
java.lang.AssertionError: expected:<34960dc2-c254-4442-89f5-b6aaa98d70c3> but was:<null>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:118)
	at org.junit.Assert.assertEquals(Assert.java:144)
	at com.hazelcast.jet.impl.connector.StreamJmsPTest.when_queue(StreamJmsPTest.java:78)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:105)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:97)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```

--- a/hazelcast-jet-core/src/test/java/com/hazelcast/jet/impl/connector/StreamJmsPTest.java
+++ b/hazelcast-jet-core/src/test/java/com/hazelcast/jet/impl/connector/StreamJmsPTest.java
@@ -63,6 +63,7 @@ public class StreamJmsPTest extends JetTestSupport {
     @Test
     public void when_queue() throws Exception {
         String queueName = randomString();
+        logger.info("using queue: " + queueName);
         initializeProcessor(queueName, true);
         String message1 = sendMessage(queueName, true);
         String message2 = sendMessage(queueName, true);
@@ -71,16 +72,25 @@ public class StreamJmsPTest extends JetTestSupport {
 
         Queue<Object> queue = outbox.queue(0);
 
-        processor.complete();
-        assertEquals(message1, queue.poll());
+        // Even though both messages are in queue, the processor might not see them
+        // because it uses `consumer.receiveNoWait()`, so if it's not yet available, it doesn't
+        // block and it should be available later.
+        // See https://github.com/hazelcast/hazelcast-jet/issues/1010
+        assertTrueEventually(() -> {
+            processor.complete();
+            assertEquals(message1, queue.poll());
+        });
         outbox.reset();
-        processor.complete();
-        assertEquals(message2, queue.poll());
+        assertTrueEventually(() -> {
+            processor.complete();
+            assertEquals(message2, queue.poll());
+        });
     }
 
     @Test
     public void when_topic() throws Exception {
         String topicName = randomString();
+        logger.info("using topic: " + topicName);
         sendMessage(topicName, false);
         initializeProcessor(topicName, false);
         sleepSeconds(1);
@@ -120,6 +130,7 @@ public class StreamJmsPTest extends JetTestSupport {
         MessageProducer producer = session.createProducer(destination);
         TextMessage textMessage = session.createTextMessage(message);
         producer.send(textMessage);
+        logger.info("sent message " + message + " to " + destinationName);
         session.close();
         connection.close();
         return message;
@@ -142,5 +153,4 @@ public class StreamJmsPTest extends JetTestSupport {
         connection.close();
         return size;
     }
-
 }

